### ISSUE TITLE
Make options for `.signAsync` optional

### ISSUE THREAD

**Issue by xlc** [polkadot-js/api#2012]
### Goal:

Update `signAsync` to make the second argument optional, and default to `{}`

### Background:

Currently this code will crash:

`const signed = await api.system.remark().signAsync(pair, options);` when `options` is `undefined`, it gives this error

```
(node:78914) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'nonce' of undefined
    at Submittable.value (/Users/xiliangchen/projects/playground/node_modules/@polkadot/api/submittable/createClass.js:59:42)
    at Submittable.value (/Users/xiliangchen/projects/playground/node_modules/@polkadot/api/submittable/createClass.js:129:99)
    at /Users/xiliangchen/projects/playground/node_modules/@polkadot/api/submittable/createClass.js:251:110
    at /Users/xiliangchen/projects/playground/node_modules/@polkadot/api/promise/Api.js:78:14
    at Submittable.signAsync (/Users/xiliangchen/projects/playground/node_modules/@polkadot/api/submittable/createClass.js:251:170)
    at /Users/xiliangchen/projects/playground/index.ts:28:33
    at step (/Users/xiliangchen/projects/playground/index.ts:33:23)
    at Object.next (/Users/xiliangchen/projects/playground/index.ts:14:53)
    at /Users/xiliangchen/projects/playground/index.ts:8:71
    at new Promise (<anonymous>)
```

### Actions:

Make second argument optional

https://github.com/polkadot-js/api/blob/e77050b9f59a0b772d1d0b8c5c1da488a7c36810/packages/api/src/submittable/createClass.ts#L72

Update interface

https://github.com/polkadot-js/api/blob/e77050b9f59a0b772d1d0b8c5c1da488a7c36810/packages/api/src/submittable/types.ts#L55

Provide appropriate unit tests.

----

@jacogr please tag this if you are happy about this.

Tags:
`bounty-awaiting-approval`
`bounty-XS`

**Comment by jacogr** [comment]
Small one, it is one of those "don't do that", which crashy, it does indicate how to fix so no immediately critical..

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 50.0 DAI (50.0 USD @ $1.0/DAI)  attached to it as part of the web3foundation fund.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/polkadot-js/api/2012/4283).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/polkadot-js/api/2012/4283).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $90,206.67 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 266 years, 6 months from now.
Please review their action plans below:


**1) [sebastiantf](https://gitcoin.co/sebastiantf) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/polkadot-js/api/2012/4283?mutate_worker_action=approve&worker=sebastiantf) | [reject worker](https://gitcoin.co/issue/polkadot-js/api/2012/4283?mutate_worker_action=reject&worker=sebastiantf))_.**

Seems a simple task. Can turn around as possible.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/polkadot-js/api/2012/4283).

**Comment by Web3Foundation** [comment]
@developerfred as you've been asked on another issue this week, please do not start working before you're approved for an issue. 

Submitting a PR before you are approved is not only discourteous to other bounty hunters but possibly could waste time if the code in question isn't up to par. 

Please, in the future, do not start working on Pull Requests before you're approved to start work, as it disrupts everyone's time involved. 

We're happy to see your involvement and excitement to work on issues in the ecosystem and we'd love to have your contributions, but please follow the protocol before submitting any further pull requests. 

Thanks, if you have any other questions feel free to reach out to me directly - chris@web3.foundation

**Comment by developerfred** [comment]
ok, Thanks @Web3Foundation.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 50.0 DAI (50.0 USD @ $1.0/DAI) has been submitted by__:

 1. [@kichjang](https://gitcoin.co/kichjang)



@web3foundation please take a look at the submitted work:
* [PR](https://github.com/polkadot-js/api/pull/2238) by @kichjang

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/polkadot-js/api/2012/4283)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/polkadot-js/api/2012/4283).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $111,500.76 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 50.0 DAI (50.0 USD @ $1.0/DAI)  attached to this issue has been approved & issued to @kichjang.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/polkadot-js/api/2012/4283)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $115,786.94 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by Web3Foundation** [comment]
Hey @KiChjang  Thanks for working on this issue. Please help us improve the W3F bounties by filling out this very short survey: https://info.polkadot.network/bounty-completion-satisfaction-survey

It'll take you a minute but can help greatly improve the program.

Thanks,
W3F
